nexus/src/external_api/http_entrypoints.rs

```diff
@@ -4023,6 +4024,97 @@ async fn updates_refresh(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// Metrics
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct ResourceUtilization {
+    #[serde(flatten)]
+    pub pagination: dropshot::PaginationParams<
+        params::ResourceMetrics,
+        params::ResourceMetrics,
+    >,
+
+    /// The UUID of the container being queried
+    // TODO: I might want to force the caller to specify type here?
+    pub id: Uuid,
+}
+
+#[derive(Display, Deserialize, JsonSchema)]
+#[display(style = "snake_case")]
+#[serde(rename_all = "snake_case")]
+pub enum ResourceName {
+    VirtualDiskSpaceProvisioned,
+    CpusProvisioned,
+    RamProvisioned,
+}
+
+#[derive(Deserialize, JsonSchema)]
+struct SystemMetricsPathParam {
+    resource_name: ResourceName,
+}
+
+/// Access metrics data
+#[endpoint {
+     method = GET,
+     path = "/system/metrics/{resource_name}",
+     tags = ["system"],
+}]
+async fn system_metrics_list(
```

Comment:
When I hear "list metrics" I think I'm listing the _available_ metrics (i.e., the schema), but I think this is returning actual data points.  Is that right?

Reply:
Agree, this threw me off. This could just be `system_metric`.

Reply:
Updated to `system_metric`: 8d153f6277e82436363181377266760c72356e50